File: library/src/main/java/com/cz/android/sample/library/component/memory/SampleMemoryComponent.java
package com.cz.android.sample.library.component.memory;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;

import com.cz.android.sample.component.ComponentContainer;
import com.cz.android.sample.library.R;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

/**
 * @author Created by cz
 * @date 2020-01-29 19:06
 * @email bingo110@126.com
 */
public class SampleMemoryComponent implements ComponentContainer {

    @Override
    public boolean isComponentAvailable(@NonNull Object object) {
        SampleMemory sampleDocument = object.getClass().getAnnotation(SampleMemory.class);
        return null!=sampleDocument&&sampleDocument.value();
    }

    @Override
    public View getComponentView(@NonNull FragmentActivity context, @NonNull Object object, @NonNull ViewGroup parentView, @NonNull View view) {
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        View createView = layoutInflater.inflate(R.layout.sample_memory_layout, parentView, false);
        final View memoryView = createView.findViewById(R.id.sampleMemoryView);
        final ViewGroup sampleMemoryContainer = createView.findViewById(R.id.sampleMemoryContainer);
        final FloatingActionButton sampleMemoryButton = createView.findViewById(R.id.sampleMemoryButton);
        sampleMemoryButton.setSelected(true);
        sampleMemoryButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                sampleMemoryButton.setSelected(!sampleMemoryButton.isSelected());
                memoryView.setVisibility(sampleMemoryButton.isSelected()?View.VISIBLE:View.INVISIBLE);
            }
        });
        sampleMemoryContainer.addView(view);
        return createView;
    }

    @Override
    public void onCreatedView(@NonNull FragmentActivity context, @NonNull Object object, @NonNull View view) {
    }

    @Override
    public int getComponentPriority() {
        return 1;
    }
}

File: extension/src/main/java/com/github/jackchen/android/sample/library/component/memory/SampleMemoryComponent.kt
package com.github.jackchen.android.sample.library.component.memory

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import com.github.jackchen.android.core.component.ComponentContainer
import com.github.jackchen.android.sample.api.Extension
import com.github.jackchen.android.sample.library.R
import com.google.android.material.floatingactionbutton.FloatingActionButton

/**
 * @author Created by cz
 * @date 2020-01-29 19:06
 * @email bingo110@126.com
 */
@Extension
class SampleMemoryComponent : ComponentContainer {
    override fun isComponentAvailable(component: Any): Boolean {
        val sampleDocument = component.javaClass.getAnnotation(SampleMemory::class.java)
        return null != sampleDocument && sampleDocument.value
    }

    override fun getComponentView(
        context: AppCompatActivity, component: Any,
        parentView: ViewGroup, view: View
    ): View {
        val layoutInflater = LayoutInflater.from(context)
        val createView = layoutInflater.inflate(R.layout.sample_memory_layout, parentView, false)
        val memoryView = createView.findViewById<View>(R.id.sampleMemoryView)
        val sampleMemoryContainer = createView.findViewById<ViewGroup>(R.id.sampleMemoryContainer)
        val sampleMemoryButton = createView.findViewById<FloatingActionButton>(R.id.sampleMemoryButton)
        sampleMemoryButton.isSelected = true
        sampleMemoryButton.setOnClickListener {
            sampleMemoryButton.isSelected = !sampleMemoryButton.isSelected
            memoryView.visibility = if (sampleMemoryButton.isSelected) View.VISIBLE else View.INVISIBLE
        }
        sampleMemoryContainer.addView(view)
        return createView
    }

    override fun onCreatedView(context: AppCompatActivity, `object`: Any, view: View) {}
}